torchtune/training/checkpointing/_checkpoint_client.py

```diff
@@ -182,20 +189,25 @@ def _save_checkpoint_async(
             # because you can only do async checkpointing one at a time
             adapter_start = time.perf_counter()
 
-            save_path = dcp_saver.get_output_path(epoch=epoch)
-            os.makedirs(save_path, exist_ok=True)
-
             dcp_saver.save_checkpoint(
                 ckpt_dict[training.ADAPTER_KEY],
                 epoch=epoch,
                 save_async=True,
                 adapter_only=True,
+                step=training_progress.steps_run,
+                dir_prefix=dir_prefix,
             )
 
             if adapter_only:
+                save_path = dcp_saver.output_dir
+                if dir_prefix == "step":
```

Comment:
Might be worth a comment that this is fairly hacky b/c we need to infer BC with epochs and potentially could be refactored